TestController unit test fails intermittently because etcd sync updates lost

The cmd/controller/main_test.go TestController test fails intermittenly with:
```
--- FAIL: TestController (31.75s)
    clusterinst_testutil.go:340:
                Error Trace:    clusterinst_testutil.go:340
                                                        clusterinst_testutil.go:260
                                                        main_test.go:133
                Error:          Expected nil, but got: &status.Error{s:(*status.Status)(0xc00072a8c8)}
                Test:           TestController
                Messages:       Create ClusterInst {"cluster_key":{"name":"Pillimos"},"cloudlet_key":{"organization":"UFGT Inc.","name":"San Jose Site"},"organization":"AtlanticInc"}
```
To reproduce, run
```
cd cmd/controller
go test -run TestController -count 30 -timeout 20m
```
I tested against both etcd 3.2.16 and etcd 3.5.4.

After debugging it turns out some sync updates were lost, so changes watched from etcd were not making into the Controller caches, and thus not being sent over notify to the fake CRM.

--- a/cmd/controller/sync.go
+++ b/cmd/controller/sync.go
@@ -139,6 +139,8 @@ func (s *Sync) syncCb(ctx context.Context, data *objstore.SyncCbData) {
 	case objstore.SyncList:
 		fallthrough
 	case objstore.SyncUpdate:
+		fallthrough
+	case objstore.SyncDelete:
 		if cache, found := s.GetCache(ctx, data.Key); found {
 			// To guarantee that objects are sent via notify
 			// in the correct order, we need to sort all object
@@ -148,16 +150,13 @@ func (s *Sync) syncCb(ctx context.Context, data *objstore.SyncCbData) {
 		}
 		if !data.MoreEvents {
 			for _, d := range s.batch {
-				d.cache.SyncUpdate(ctx, d.data.Key, d.data.Value, d.data.Rev, d.data.ModRev)
+				if d.data.Action == objstore.SyncUpdate || d.data.Action == objstore.SyncList {
+					d.cache.SyncUpdate(ctx, d.data.Key, d.data.Value, d.data.Rev, d.data.ModRev)
+				} else if d.data.Action == objstore.SyncDelete {
+					d.cache.SyncDelete(ctx, d.data.Key, d.data.Rev, d.data.ModRev)
+				}
 			}
 			s.batch = nil
-			s.rev = data.Rev
-		}
-	case objstore.SyncDelete:
-		if cache, found := s.GetCache(ctx, data.Key); found {
-			cache.SyncDelete(ctx, data.Key, data.Rev, data.ModRev)
-		}
-		if !data.MoreEvents {
 			s.rev = data.Rev
 		}
 	}

